cmd/oci-unpack/main.go

```diff
@@ -95,13 +101,18 @@ func (v *unpackCmd) Run(cmd *cobra.Command, args []string) {
 		v.typ = typ
 	}
 
+	isUserRoot := os.Getuid() == 0
+	unpacker := &image.Unpacker{
+		PreserveOwnership: v.sameOwner || isUserRoot,
```

Comment:
This `isUserRoot` check can happen when you setup the `BoolVar` default, which would keep it right next to the documentation for the conditional behavior.